plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/Inline.hs

```diff
@@ -190,27 +195,37 @@ processTerm = handleTerm <=< traverseOf termSubtypes applyTypeSubstitution where
             t' <- processTerm t
             -- Use 'mkLet': we're using lists of bindings rather than NonEmpty since we might
             -- actually have got rid of all of them!
-            pure $ mkLet a NonRec bs' t'
+            pure $ mkLet ann NonRec bs' t'
         -- This includes recursive let terms, we don't even consider inlining them at the moment
         t -> forMOf termSubterms t processTerm
```

Comment:
```suggestion
        t -> do
             t' <- tryInlineSat t
             forMOf termSubterms t' processTerm
```
Writing it like this exposes an important subtlety that we have to think about: do we want to continue processing the term after we've inlined the head of an application? i.e. in which order do we want to run these steps? It won't affect our inlining decision here, but it will affect how much work we do. This also depends on whether or not we have processed the RHS of the binding that we put into the `VarInfo`. See p11 of the paper for why it's important.

Reply:
Okay, so the RHS in the `VarInfo` is processed. Then we should first of all process the subterms of the application and _then_ consider saturated inlining, since otherwise we will re-process the RHS.